scripts/numa_prefix

```diff
@@ -23,45 +23,54 @@
 #       1:  20  10
 #============================================================================
 
-use strict;
-use warnings;
+import subprocess
+import re
+import sys
 
-my $path = `which numactl`;
-if(length($path) > 0) {
-  my ($head_line, @rest) = map {chomp; $_} `numactl -H`;
+which_proc = subprocess.Popen(["which", "numactl"], stdout=subprocess.PIPE)
+out, err = which_proc.communicate()
 
-  if($head_line =~ /available: (\d+) nodes/) {
-    my $node_count     = $1;
-    my $best_node_id   = undef
-    my $best_cpus      = undef;
-    my $best_free_size = undef;
+if out != "":
+    numactl_proc = subprocess.Popen(["numactl", "-H"], stdout=subprocess.PIPE)
+    out, err = numactl_proc.communicate()
 
-		# loop through available nodes, selecting the node with the most free mem
-    foreach my $num (1..$node_count) {
-      my $cpus_line     = shift(@rest);
-      my $mem_size_line = shift(@rest);
-      my $mem_free_line = shift(@rest);
+    lines = out.split("\n")
+    line_idx = 0
 
-      if($cpus_line =~ /node (\d+) cpus: (\d.*\d)$/) {
-        my ($node_id, $cpus) = ($1, $2);
-        $cpus =~ s/\s+/,/g;
+    head_line = lines[line_idx]
+    line_idx += 1
+    node_match = re.match(r"^ *available: +(\d+) nodes", head_line)
+    if node_match:
+        avail_nodes = node_match.group(1)
+        best_node_id = ""
+        best_cpus = ""
+        best_free_size = 0
 
-        if($mem_free_line =~ /node $node_id free: (\d+) \S+$/) {
-          my $free_size = $1;
-          if(!defined($best_free_size) || ($free_size > $best_free_size)) {
-            $best_node_id   = $node_id;
-            $best_cpus      = $cpus;
-            $best_free_size = $free_size;
-          }
-        } else {
-          die("malformed mem-free line: $mem_free_line\n");
-        }
-      } else {
-        die("malformed cpus line: $cpus_line\n");
-      }
-    }
-    print("numactl -m $best_node_id -C $best_cpus --");
-  } else {
-    die("malformed head line: $head_line\n");
-  }
-}
+        # loop through available nodes, selecting the node with the most free mem
+        for i in avail_nodes:
+            cpu_line      = lines[line_idx]
+            mem_size_line = lines[line_idx + 1]
```

Comment:
This isn't used anywhere. I think we should just remove it and comment on mem_free_line that we skip the size line.